Adyen/Core/APIClient/UniqueAssetAPIClient.swift

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// :nodoc:
+/// An API Client that fetches an asset only once.
+public final class UniqueAssetAPIClient<ResponseType: Response> {
+
+    /// :nodoc:
+    public typealias CompletionHandler = (Result<ResponseType, Error>) -> Void
+
+    /// :nodoc:
+    private let apiClient: APIClientProtocol
+
+    /// :nodoc:
+    private var waitingList: [CompletionHandler] = []
+
+    /// :nodoc:
+    private var cachedResponse: ResponseType?
```

Comment:
Yo can "go back" on VoucherComponent, right?
Maybe this should be a dictionary? 
AFAIU right now you can only generate one Voucher per VoucherActionComponent...

Reply:
when you cancel the voucher component or click finish the apple wallet provider is nilled, and regenerated when you click submit again

https://github.com/Adyen/adyen-ios/blob/cb0f4ed610c5128ce5629c93050011a5686bad74/AdyenActions/Components/Voucher/VoucherComponent.swift#L76

https://github.com/Adyen/adyen-ios/blob/cb0f4ed610c5128ce5629c93050011a5686bad74/AdyenActions/Components/Voucher/VoucherComponent.swift#L122

Reply:
👍